src/odemis/driver/ephemeron.py

```diff
@@ -0,0 +1,747 @@
+             DataType=NodeId(ObjectIds.Float),
+             ValueRank=-1,
+             ArrayDimensions=[],
+             Description=LocalizedText("Delay time (ms)")),
+    Argument(Name="points_fast",
+             DataType=NodeId(ObjectIds.Int64),
+             ValueRank=-1,
+             ArrayDimensions=[],
+             Description=LocalizedText("Points Fast")),
+    Argument(Name="points_slow",
+             DataType=NodeId(ObjectIds.Int64),
+             ValueRank=-1,
+             ArrayDimensions=[],
+             Description=LocalizedText("Points Slow")),
+    Argument(Name="simulate",
+             DataType=NodeId(ObjectIds.Boolean),
+             ValueRank=-1,
+             ArrayDimensions=[],
+             Description=LocalizedText("Simulate Scan")),
+]
+
+DWELLTIME_ARGS = [
+    Argument(Name="oversampling",
+             DataType=NodeId(ObjectIds.Int64),
+             ValueRank=-1,
+             ArrayDimensions=[],
+             Description=LocalizedText("Oversampling Value")),
+    Argument(Name="channels",
+             DataType=NodeId(ObjectIds.Int64),
+             ValueRank=-1,
+             ArrayDimensions=[],
+             Description=LocalizedText("Number of Scan Channels")),
+    Argument(Name="samples",
+             DataType=NodeId(ObjectIds.Int64),
+             ValueRank=-1,
+             ArrayDimensions=[],
+             Description=LocalizedText("Number of Samples")),
+    Argument(Name="delay",
+             DataType=NodeId(ObjectIds.Float),
+             ValueRank=-1,
+             ArrayDimensions=[],
+             Description=LocalizedText("Delay time (ms)"))
+]
+
+SCANTIME_ARGS = [
+    Argument(Name="dwell_time",
+             DataType=NodeId(ObjectIds.Int64),
+             ValueRank=-1, ArrayDimensions=[],
+             Description=LocalizedText("Dwell Time (ms)")),
+    Argument(Name="points_fast",
+             DataType=NodeId(ObjectIds.Int64),
+             ValueRank=-1,
+             ArrayDimensions=[],
+             Description=LocalizedText("Points Fast")),
+    Argument(Name="points_slow",
+             DataType=NodeId(ObjectIds.Int64),
+             ValueRank=-1,
+             ArrayDimensions=[],
+             Description=LocalizedText("Points Slow"))
+]
+
+STATE_ARGS = [
+    Argument(Name="new_state",
+             DataType=NodeId(ObjectIds.String),
+             ValueRank=-1,
+             ArrayDimensions=[],
+             Description=LocalizedText("Requested State"))
+]
+
+
+class MightyEBIC(Detector):
+    def __init__(self, name: str, role: str, channel: int, url: str, namespace: str, **kwargs):
+        """
+        Initialise the EBIC controller
+        :param name (str): The name of the device configured through the configuration file
+        :param role (str): The role of the device configured through the configuration file
+        :param channel (int): The channel the device should use for hw triggering.
+        :param url (str): The url address to use with the OPC UA protocol
```

Comment:
Please indicate a special string for running the simulator. It shouldn't be a standard URL, but more like "simulator" or "fake". If the code sees this, it should replace it by a hard-coded URL that points to localhost, and start the simulator.
This is more explicit for the user, but also allows to have the mighty EBIC server run on the same computer as Odemis, without being confused as a simulator.